addons/website/tests/test_website_visitor.py

```diff
@@ -324,3 +325,113 @@ def test_visitor_archive(self):
         new_visitor = self.env['website.visitor'].search([('id', '=', new_visitor.id)])
         self.assertEqual(len(new_visitor), 0, "The new visitor should be deleted when visitor authenticate once again.")
         self.assertEqual(old_visitor.active, True, "The old visitor should be reactivated when visitor authenticates once again.")
+
+@tagged('-at_install', 'post_install')
+class TestPortalWizardMultiWebsites(HttpCase):
+    def test_portal_wizard_multi_websites(self):
+        website1 = self.env['website'].create({
+            'name': 'website1_specific_user_account',
+            'specific_user_account': True,
+        })
+        website2 = self.env['website'].create({
+            'name': 'website2_specific_user_account',
+            'specific_user_account': True,
+        })
+        website1_shared_user_account = self.env['website'].create({
+            'name': 'website1_shared_user_account',
+        })
+        website2_shared_user_account = self.env['website'].create({
+            'name': 'website2_shared_user_account',
+        })
+        partner_1 = self.env['res.partner'].create({
+            'name': 'partner_1',
+            'email': 'email_address@example.com',
+            'website_id': website1.id,
+        })
+        partner_1_same_website = self.env['res.partner'].create({
+            'name': 'partner_1_same_website',
+            'email': 'email_address@example.com',
+            'website_id': website1.id,
+        })
+        partner_2 = self.env['res.partner'].create({
+            'name': 'partner_2',
+            'email': 'email_address@example.com',
+            'website_id': website2.id,
+        })
+        partner_shared_user_account = self.env['res.partner'].create({
+            'name': 'partner_shared_user_account',
+            'email': 'email_address@example.com',
+            'website_id': website1_shared_user_account.id,
+        })
+        partner_shared_user_account_different_website = self.env['res.partner'].create({
+            'name': 'partner_shared_user_account_different_website',
+            'email': 'email_address@example.com',
+            'website_id': website2_shared_user_account.id,
+        })
+        portal_wizard_1 = self.env['portal.wizard'].with_context(active_ids=[partner_1.id]).create({})
+        self.assertEqual(len(portal_wizard_1.user_ids), 1)
```

Comment:
Is this check really needed as the grant access do a `self.ensure_one()` ? Same everywhere in this test.

Reply:
Indeed, I removed it :+1: